integrations/jira/jira/client.py

```diff
@@ -44,24 +55,93 @@ def __init__(self, jira_url: str, jira_email: str, jira_token: str) -> None:
             "maxResults": maxResults,
             "startAt": startAt,
         }
 
-    async def _get_paginated_projects(self, params: dict[str, Any]) -> dict[str, Any]:
-        project_response = await self.client.get(
-            f"{self.api_url}/project/search", params=params
-        )
-        project_response.raise_for_status()
-        return project_response.json()
+    async def _make_paginated_request(
+        self,
+        url: str,
+        params: dict[str, Any] = {},
+        data_key: str = "values",
+        is_last_function: typing.Callable[
+            [dict[str, Any]], bool
+        ] = lambda response: response["isLast"],
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = {**self._generate_base_req_params(), **params}
+        is_last = False
+        logger.info(f"Making paginated request to {url} with params: {params}")
+        while not is_last:
+            try:
+                response = await self.client.get(url, params=params)
+                response.raise_for_status()
+                response_data = response.json()
+                values = response_data[data_key]
+                yield values
+                is_last = is_last_function(response_data)
+                start = response_data["startAt"] + response_data["maxResults"]
+                params = {**params, "startAt": start}
+                logger.info(f"Next page startAt: {start}")
+            except httpx.HTTPStatusError as e:
+                logger.error(
+                    f"HTTP error with status code: {e.response.status_code} and response text: {e.response.text}"
+                )
+                raise
+        logger.info("Finished paginated request")
+        return
+
+    async def get_projects(self) -> AsyncGenerator[list[dict[str, Any]], None]:
+        async for projects in self._make_paginated_request(
+            f"{self.detail_base_url}/project/search"
+        ):
+            yield projects
+
+    async def get_issues(
+        self, board_id: int
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        params = {}
+        config = typing.cast(JiraResourceConfig, event.resource_config)
 
-    async def _get_paginated_issues(self, params: dict[str, Any]) -> dict[str, Any]:
-        issue_response = await self.client.get(f"{self.api_url}/search", params=params)
-        issue_response.raise_for_status()
-        return issue_response.json()
+        if config.selector.jql:
+            params["jql"] = config.selector.jql
+            logger.info(f"Found JQL filter: {config.selector.jql}")
+
+        async for issues in self._make_paginated_request(
+            f"{self.base_url}/board/{board_id}/issue",
+            params=params,
+            data_key="issues",
+            is_last_function=lambda response: response["startAt"]
+            + response["maxResults"]
+            >= response["total"],
+        ):
+            yield [{**issue, "boardId": board_id} for issue in issues]
```

Comment:
We have a convention where we distinguish additional properties by using double underscore. so this will be 

` yield [{**issue, "__boardId": board_id} for issue in issues]`